### ISSUE TITLE
IP check fails for my own public IP

### ISSUE THREAD

**Issue by michaeldzjap** [Torann/laravel-geoip#128]
I am trying out _Laravel Geoip_ with my own public IP address, but the `isValid()` method of `GeoIP` is returning `false` for some reason, thus bypassing the whole geolocation lookup. I have looked at the definition of `isValid()` and I have to admit that I am a little confused about its logic checks. In my case 

```php
filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)
```

returns my IP and  

```php
filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_NO_PRIV_RANGE)
```

returns `false`. So I suppose I may conclude that my IP is a valid IPV4 address. What is the purpose of negating each condition and then requiring they are both truthy?

I've also tried to bypass the whole `isValid()` check so that the request to my geolocation service is actually made (I am using ipstack through a custom service) and this returns a valid response with acceptable geolocation information.

**Comment by Torann** [comment]
Sounds like you might be testing you local IP and not your public facing one. `FILTER_FLAG_NO_RES_RANGE` fails validation for the following reserved IPv4 ranges: 0.0.0.0/8, 169.254.0.0/16, 127.0.0.0/8 and 240.0.0.0/4 and the following reserved IPv6 ranges: ::1/128, ::/128,::ffff:0:0/96 and fe80::/10.
--

**Comment by michaeldzjap** [comment]
> Sounds like you might be testing you local IP and not your public facing one. 

No, I am certain that I was using my public IP address. Well, at least the public IP address _Google_ returns when you search for "my IP". I assume that this is the correct public IP address for my machine. However, I decided not to use this library in the end, so this issue can be closed as far as I am concerned.

**Comment by juhasev** [comment]
I have a same issue and noticed while debugging the code that we have two filters here:
```
$firstCondition = !filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE);

$secondCondition = !filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_NO_PRIV_RANGE);
```
IP address given here is: 172.19.0.1

Next we have a condition:
```
 if ($firstCondition && $secondCondition) {
            return false;
 }
```
And this is where it go bad obviously the IP passed both filters which it should but then we are returning false??? This should be instead:
 
```
if (!$firstCondition || !$secondCondition) {
            return false;
 }
```

**Comment by muffaddal** [comment]
Can we test different country Ips through VPN? I have a module where I have to send data on the basis of geoip. i want to test this from different country ips.

**Comment by srakl** [comment]
How did you get this to work with ipstack? what changes need to be made? michaeldzjap](https://github.com/michaeldzjap)
